### PR TITLE
Fix don't overwrite the span status of unfinished spans

### PR DESCRIPTION
**Pull request by markushi**
## :scroll: Description
Span status should remain untouched when being force-finished, instead of setting `DEADLINE_EXCEEDED`.

## :bulb: Motivation and Context
Fixes https://github.com/getsentry/sentry-java/issues/2730

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry/src/main/java/io/sentry/SentryTracer.java [hunk 1]
```diff
@@ -198,7 +198,7 @@ public void finish(
         if (!child.isFinished()) {
           child.setSpanFinishedCallback(
               null); // reset the callback, as we're already in the finish method
-          child.finish(SpanStatus.DEADLINE_EXCEEDED, finishTimestamp);
+          child.finish(child.getStatus(), finishTimestamp);
```
Comment (romtsn) [inline sentry/src/main/java/io/sentry/SentryTracer.java:201]: hm, but as I understand, we don't even need this logic anymore at all (iterating through children and finishing them), relay will just do that for us?

Comment (romtsn) [reply sentry/src/main/java/io/sentry/SentryTracer.java:201]: image

Comment (markushi) [reply sentry/src/main/java/io/sentry/SentryTracer.java:201]: Oh right, good catch. Also: the status only gets set if there's no end timestamp: https://github.com/getsentry/relay/pull/1690/files#diff-7f49c53574a8b3eef93c44e6df72f927d591b4f00ebf791af696ea6eae77fb2bR280-R283
Let me fix that

**Review by romtsn — APPROVED** [review]
LGTM, but only one concern - we're not executing this block for JPC stuff, do we still need to handle it (at least the start date), in case of parent transaction is aborted?

https://github.com/getsentry/sentry-java/blob/695d3a3acd91e7759da0699f9b5e004d945de53b/sentry/src/main/java/io/sentry/Span.java#L193-L222

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,7 @@ Breaking changes:
 
 Breaking changes:
 - Move enableNdk from SentryOptions to SentryAndroidOptions ([#2793](https://github.com/getsentry/sentry-java/pull/2793))
+- Fix don't overwrite the span status of unfinished spans ([#2859](https://github.com/getsentry/sentry-java/pull/2859))
```
Comment (romtsn) [inline CHANGELOG.md:16]: Also, should we mention min support for self-hosted? as per https://github.com/getsentry/sentry-java/issues/2730#issuecomment-1569843157

**Review by stefanosiano — APPROVED** [review]
good!

**Pull request merged**